openedx/stanford/cms/djangoapps/contentstore/views/utilities/bulkupdate.py

```diff
@@ -0,0 +1,257 @@
+            'max_attempts': 'fillerMaxAttempts',
+            'show_answer': 'fillerShowAnswer'
+        }
+    )
+
+    course_key = CourseKey.from_string(course_key_string)
+    course = modulestore().get_course(course_key, 3)
+
+    return render_to_response(
+        'bulkupdate.html',
+        {
+            'context_course': course,
+            'course_outline_url': course_outline_url,
+            'bulkupdate_update_url': update_url,
+            'bulkupdate_status_url': status_url,
+            'default_max_attempts': DEFAULT_MAX_ATTEMPTS,
+            'default_show_answer': DEFAULT_SHOW_ANSWER,
+            'show_answer_options': SHOW_ANSWER_OPTIONS,
+        }
+    )
+
+
+def _utility_bulkupdate_post_handler(request, course_key_string):
+    """
+    Internal bulkupdate handler for POST operation
+    """
+    modified_settings = {}
+    user = request.user
+
+    # Retrieve settings from request
+    try:
+        max_attempts = request.POST.get('maxAttempts')
+        show_answer = request.POST.get('showAnswer')
+        session_status = request.session.setdefault("update_status", {})
+        session_status_string = _get_session_status_string(course_key_string, max_attempts, show_answer)
+    except:
+        return JsonResponse(
+            {
+                'ErrMsg': 'Missing or invalid arguments',
+                'Stage': -1
+            },
+            status=400,
+        )
+
+    # Validate settings
+    _save_request_status(request, session_status_string, 1)
+
+    if max_attempts and max_attempts != 'null':
+        try:
+            max_attempts = int(max_attempts)
+        except ValueError:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for max attempts is not an integer',
+                    'Stage': -1
+                },
+                status=400,
+            )
+        if max_attempts < 0:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for max attempts is negative',
+                    'Stage': -1
+                },
+                status=400,
+            )
+        else:
+            modified_settings['max_attempts'] = max_attempts
+
+    if show_answer and show_answer != 'null':
+        if show_answer in SHOW_ANSWER_OPTIONS:
+            modified_settings['showanswer'] = show_answer
+        else:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for show answer is not an available option',
+                    'Stage': -1
+                },
+                status=400,
+            )
+    _save_request_status(request, session_status_string, 2)
+    _update_advanced_settings(request, course_key_string, session_status_string, modified_settings)
```

Comment:
add delay to run on celery.  Make sure order of the two celery tasks doesn't matter.